src/modules/languages/rust.nix

```diff
@@ -122,6 +123,56 @@ in
       defaultText = lib.literalExpression "nixpkgs";
       description = "Rust component packages. May optionally define additional components, for example `miri`.";
     };
+
+    dev = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Enable Rust development tools.";
+      };
+
+      lsp = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = true;
+          description = "Enable rust-analyzer language server.";
```

Comment:
@domenkozar : Its tedious I know, but I suggest that these documentation strings are always kind of generic and do not mention the implementation. I made some suggestions in the `p`ython.nix`. That would be cool and more streamlined.